### DISCUSSION TITLE
TypeError ArrayBufferView

### DISCUSSION THREAD

**Discussion by HolgerBuick** [pixijs/pixijs#8355] [Q&A] [answered]
Hello all,
this code I extracted from my project.
I like to get some single pixels to the sprite.
Opened this code in Chrome and Edge and got the error message.
But  **this.data** in line 382 (parameter 9)  **is** Uint8Array, what I can see in the browser debugger and Uint8Array is an ArrayBufferView  [https://developer.mozilla.org/en-US/docs/Web/API/ArrayBufferView](url)

Any suggestions?
Thanks in advance! 

```javascript
<head>
	<title>test</title>
</head>
<script src="https://pixijs.download/v6.3.0/pixi.js"></script>

<body>
	<script>
	const app = new PIXI.Application({
		width: window.innerWidth,
		height: window.innerHeight,
		backgroundColor: 0x2c3e50
	});
	let ar = new Uint8Array([1, 2, 3, 4]);
	const options = {
		width: 2,
		height: 2
	};
	let buf = new PIXI.BufferResource(ar, options);
	let texture = new PIXI.Texture.fromBuffer(buf, 2, 2);
	let ps = new PIXI.Sprite(texture);
	app.stage.addChild(ps);
	</script>
</body>
<script>
```

382	 gl.texImage2D(baseTexture.target, 0, glTexture.internalFormat, width, height, 0, baseTexture.format, glTexture.type, this.data);
  
       Uncaught TypeError: Failed to execute 'texImage2D' on 'WebGLRenderingContext': parameter 9 is not of type 'ArrayBufferView'.
    at BufferResource.upload (core.js:382:16)
    at TextureSystem.updateTexture (core.js:8750:50)
    at TextureSystem.bind (core.js:8635:22)
    at BatchPlugin.AbstractBatchRenderer.bindAndClearTexArray (core.js:10102:27)
    at BatchPlugin.AbstractBatchRenderer.drawBatches (core.js:10138:22)
    at BatchPlugin.AbstractBatchRenderer.flush (core.js:10157:14)
    at Renderer.render (core.js:9594:36)
    at Application.render (app.js:92:23)
    at TickerListener.emit (ticker.js:97:25)
    at Ticker.update (ticker.js:481:37),

**Comment by ivanpopelyshev** [comment]
if you use BufferResource, you can pass it to `new PIXI.Texture(new PIXI.BaseTexture(buf))`

`fromBuffer` is a shortcut for that that takes `ar`

**Reply by HolgerBuick** [reply]
It results in this message:

core.js:382 
        
       WebGL: INVALID_OPERATION: texImage2D: ArrayBufferView not big enough for request

test2.html:1 
        
       [.WebGL-00005AD80039DB00] GL_INVALID_OPERATION: Texture format does not support mipmap generation.

**Reply by ivanpopelyshev** [reply]
4 pixels, each has 4 channels by one byte = 16 bytes. You have only 4

**Comment by bhousel** [comment]
I think `new PIXI.Texture.fromBuffer` can just accept the `Uint8Array` directly..

Does `new PIXI.Texture.fromBuffer(ar, 2, 2);` work?

**Reply by HolgerBuick** [reply]
It results in this message:

core.js:382 
        
       WebGL: INVALID_OPERATION: texImage2D: ArrayBufferView not big enough for request
BufferResource.upload @ core.js:382
test2.html:1 
        
       [.WebGL-00005AD80039E900] GL_INVALID_OPERATION: Texture format does not support mipmap generation.

**Comment by bigtimebuddy** [comment, marked as answer]
Here's an example: https://jsfiddle.net/bigtimebuddy/g498u3x5/

**Reply by HolgerBuick** [reply]
This example will make it!
